### PR TITLE
Add base task for auto-retrys on known exceptions

### PR DESCRIPTION
**Pull request by bbaker6225**
ENT-3594

Example:
image

### CODE REVIEW

File: enterprise_catalog/apps/api/tasks.py [hunk 1]
```diff
@@ -87,7 +89,25 @@ def _fetch_courses_by_keys(course_keys):
...
-@shared_task(base=LoggedTask)
+class LoggedTaskWithRetry(LoggedTask):  # pylint: disable=abstract-method
+    """
+    Shared base task that allows tasks that raise some common exceptions to retry automatically.
+
+    See https://docs.celeryproject.org/en/stable/userguide/tasks.html#automatic-retry-for-known-exceptions for
+    more documentation.
+    """
+    autoretry_for = (
+        IntegrityError,
```
Comment (kellmarie) [inline enterprise_catalog/apps/api/tasks.py:100]: Why does retrying on an IntegrityError work? Is there sometimes a background change with the db that just needs to finish for the task succeed on a retry?

**Review by chavesj — APPROVED** [review]
nice :shipit:

File: enterprise_catalog/apps/api/tasks.py (hunk 1 above)
Comment (bbaker6225) [reply enterprise_catalog/apps/api/tasks.py:100]: Our theory is that the integrity errors occasionally pop up when two different processes are trying to create the same ContentMetadata object at the same time. Retrying (manually) typically fixes the issue and what I found online suggested that transactions in celery tasks (especially long ones) are an expensive and not especially common thing

Comment (kellmarie) [reply enterprise_catalog/apps/api/tasks.py:100]: Makes sense to me.

**Review by kellmarie — APPROVED** [review]

**Review by adamstankiewicz — APPROVED** [review]

**Review by iloveagent57 — APPROVED** [review]

**Pull request merged**